client/src/thunks.ts

```diff
@@ -208,3 +217,15 @@ export const thunkChangePageSize: AppThunkActionCreator = (size: PageSize) => {
     dispatch(thunkFetchBits());
   };
 };
+
+export const thunkSetStateFromUrlBar: AppThunkActionCreator = () => {
+  return async (dispatch, getState) => {
+    dispatch(setPageSize(getPageSize(history.location.search)));
```

Comment:
i kind of prefer

```typescript
const urlParams = parseUrlParams(history.location);
if (urlParams.pageSize) {
  dispatch(setPageSize(urlParams.pageSize));
}
```

but this works too

Reply:
Well you're right to check that urlParams.pageSize is defined first -- turns out that's actually a bug here that I'll send a PR for momentarily. The one method to parse all URL params is probably a good idea, but going to add a TODO to refactor that.